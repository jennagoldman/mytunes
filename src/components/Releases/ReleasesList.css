ul {
  list-style: none;
}

.releaseListItem {
  display: inline-block;
}