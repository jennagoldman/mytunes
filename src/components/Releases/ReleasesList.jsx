import React from 'react';
import { Link } from 'react-router-dom';
import ReleaseItem from './ReleaseItem.jsx';
import PropTypes from 'prop-types';
import styles from './ReleasesList.css';

const ReleasesList = ({ artist, releases }) => {
  const releasesListItems = releases.map(release => (
    <li className={styles.releaseListItem} key={release.releaseId}>
      <Link to={`/release/${artist.name}/${release.releaseId}`}>
        <ReleaseItem {...release} />
      </Link>
    </li>
  ));

  return (
    <>
      <h2>{artist.name} - Release History</h2>
      <ul>
        {releasesListItems}
      </ul>
    </>
  );
};

ReleasesList.propTypes = {
  artist: PropTypes.object.isRequired,
  releases: PropTypes.array.isRequired,
  onBrokenImage: PropTypes.func.isRequired
};

export default ReleasesList;
