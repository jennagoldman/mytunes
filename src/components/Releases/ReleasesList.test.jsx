import React from 'react';
import { shallow } from 'enzyme';
import ReleasesList from './ReleasesList.jsx';

describe('ReleasesList', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<ReleasesList 
      artist={{ name: 'Electric Light Orchestra' }}
      releases={[
        { 
          releaseId: '123abc', 
          releaseTitle: 'Out of the Blue', 
          coverArt: true 
        }, 
        { 
          releaseId: '456def', 
          releaseTitle: 'Face the Music', 
          coverArt: true 
        }
      ]} 
      onBrokenImage={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
