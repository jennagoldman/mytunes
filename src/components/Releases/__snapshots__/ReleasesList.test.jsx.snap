// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleasesList matches a snapshot 1`] = `
<Fragment>
  <h2>
    Electric Light Orchestra
     - Release History
  </h2>
  <ul>
    <li
      className="releaseListItem"
      key="123abc"
    >
      <Link
        to="/release/Electric Light Orchestra/123abc"
      >
        <ReleaseItem
          coverArt={true}
          releaseId="123abc"
          releaseTitle="Out of the Blue"
        />
      </Link>
    </li>
    <li
      className="releaseListItem"
      key="456def"
    >
      <Link
        to="/release/Electric Light Orchestra/456def"
      >
        <ReleaseItem
          coverArt={true}
          releaseId="456def"
          releaseTitle="Face the Music"
        />
      </Link>
    </li>
  </ul>
</Fragment>
`;
